### PR TITLE
This removes all of the code used for the old ci and for the truffle

### PR DESCRIPTION
**Pull request by mischat**
This PR will update the docs and removed the old CI infrastructure. 

I am sure that the docs can do with another pass, but I believe this is a step in the right direction...

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,14 @@
...
-/vendor
 /artifacts
 /cache
 /typechain
 /abi
 
 # Secrets
-.secret
 .env
-.env.*
```
Comment (zajck) [inline .gitignore:20]: I would keep that in (I have some other `.env.*` files beside `.env.example`, and make `.env.example` as exception.

But if others are happy with this change, I'll just update my `.git/info/exclude`

File: README.md [hunk 2]
```diff
@@ -38,55 +34,46 @@ tools installed.
 
 At a minimum, you'll need:
 * Node (12.20)
-* NPM (7)
-* Ruby (2.7)
-* Bundler (> 2)
+* NPM (7.15)
```
Comment (zajck) [inline README.md:37]: Maybe we should add also max versions. Me and @hswopeams updated npm to 8.* as some point and our package-lock.json became "uncompatible" (not really uncompatibile but it was being modified every run)

File: package.json [hunk 3]
```diff
@@ -37,7 +37,7 @@
         "!artifacts/contracts/interfaces/**/*.dbg.json"
     ],
     "scripts": {
-        "contracts:run": "hardhat compile && hardhat deploy",
+        "contracts:run": "hardhat compile --force && hardhat deploy",
```
Comment (zajck) [inline package.json:40]: Would it be beneficial to also have `npx hardhat clean` in this command?

File: .gitignore (hunk 1 above)
Comment (hswopeams) [reply .gitignore:20]: I agree we should keep this in.

File: README.md (hunk 2 above)
Comment (hswopeams) [reply README.md:37]: Agree with this comment.

File: package.json (hunk 3 above)
Comment (hswopeams) [reply package.json:40]: I agree. With constant branch switching, I often run into problems and have to run` npx hardhat clean`. It's imperative to do `npx hardhat clean` and `npx hardhat compile --force before` a deploy. Both should be added to all of the `contracts:migrate` npm scripts

File: README.md [hunk 4]
```diff
@@ -38,55 +34,46 @@ tools installed.
...
+*Note that*: This command starts up built-in Hardhat Network and migrates all contracts to the Hardhat Network instance. The `.env` file has a hard-coded value for the `BOSON_TOKEN` address, which points to account #9 of the local hardhat network. This isn't an actual BOSON token address but just a valid address that will allow the deployment scripts to work. It's not necessary to deploy a BOSON token to run the unit tests locally (see Unit Tests section), as the unit tests deploy their own contract instances.
 
-If preferred by those who are familiar with Hardhat, the standard Hardhat commands can be used. Ganache can be started up manually by configuring a local network to be run against or using the `hardhat-ganache` plugin or you could start a Hardhat Network using `npx hardhat node`. For more information on how this can be achieved refer to the [official Hardhat documentation](https://hardhat.org/guides/ganache-tests.html#running-tests-with-ganache)
+If preferred by those who are familiar with Hardhat, the standard Hardhat commands can be used.
 
 In a separate terminal, contracts can be deployed using
```
Comment (hswopeams) [inline README.md:78]: If you just want to deploy to the built-in hardhat node `--network localhost` is not needed. You should only have to run` npx hardhat deploy`. A separate terminal and `--network localhost` is only needed if the `npx hardhat node` command is used.

File: README.md [hunk 5]
```diff
@@ -101,16 +88,16 @@ The following hardhat commands will achieve this:
...
+
+You can then deploy from a separate terminal using the command:
 
 ```shell script
 npx hardhat deploy --network localhost
 ```
 
-This makes the BOSON test token deployed on Rinkeby (0xEDa08eF1c6ff51Ca7Fd681295797102c1B84606c) and the official DAI token 
-deployed on Rinkeby (0x6A9865aDE2B6207dAAC49f8bCba9705dEB0B0e6D) available to your local hardhat chain.
+This makes the BOSON test token deployed on Rinkeby (0xEDa08eF1c6ff51Ca7Fd681295797102c1B84606c) and the official DAI token deployed on Rinkeby (0x6A9865aDE2B6207dAAC49f8bCba9705dEB0B0e6D) available to your local hardhat chain.
```
Comment (hswopeams) [inline README.md:100]: Doe we still want to refer to our Rinkeby deployments, or should we refer to our Ropsten deployments?

File: README.md [hunk 6]
```diff
@@ -151,11 +128,11 @@ provide test coverage reports.
 To check the test coverage: 
 
 ```shell script 
-./go tests:coverage
+npm run tests:coverage
 ```
 
-`solidity-coverage` runs its own instance of Ganache internally, as well as
-instrumenting contracts before running.
+`solidity-coverage` runs its own instance of Ganache internally, as well as instrumenting contracts before running.
```
Comment (hswopeams) [inline README.md:134]: Remove reference to Ganache. This is only applicable when using truffle. According to the docs (https://github.com/sc-forks/solidity-coverage), it runs on directly on the HardhatEVM when hardhat is used.

**Review by hswopeams — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -38,55 +34,46 @@ tools installed.
...
-```
+npm run contracts:compile
+````
 
 ---
 ### Run
-To deploy instances of the contracts for local development without prior knowledge of Hardhat, first copy .env.example to .env and run the following command:
+
 ```shell
 ./go contracts:run
```
[the remark this answers is not in the document]
Comment (hswopeams) [reply README.md:71]: It seems the` npm install` issue was a Mac-specific issue.

File: .gitignore (hunk 1 above)
Comment (mischat) [reply .gitignore:20]: done

File: README.md (hunk 2 above)
Comment (mischat) [reply README.md:37]: this is now set in package.json

File: package.json (hunk 3 above)
Comment (mischat) [reply package.json:40]: done

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (zajck) [reply README.md:71]: Beside changing ethers to 5.4.7 there are some changes in package-lock.json.
I assume it happened when I was testing out different ethers version and some dependencies changed, and after settling with 5.4.7, those dependencies in package-lock remained there.
So there were no other intentional changes beside changing ethers version.

[the remark this answers is not in the document]
Comment (mischat) [reply README.md:71]: I have updated the reference to the old go script

File: README.md (hunk 4 above)
Comment (mischat) [reply README.md:78]: thanks, made some changes to the doc

File: README.md (hunk 5 above)
Comment (mischat) [reply README.md:100]: I left them in here, and then thought we could update the docs when we deploy 1.0.0, and then we can point to instances on all three of the networks.

File: README.md (hunk 6 above)
Comment (mischat) [reply README.md:134]: updated.

**Comment by mischat** [comment]
@hswopeams i have updated the branch based on your comments ...

File: .gitignore (hunk 1 above)
Comment (hswopeams) [reply .gitignore:20]: OK for me.

File: README.md (hunk 2 above)
Comment (hswopeams) [reply README.md]: OK for me.

File: package.json (hunk 3 above)
Comment (hswopeams) [reply package.json]: OK for me.

File: README.md (hunk 5 above)
Comment (hswopeams) [reply README.md:102]: OK

**Review by hswopeams — APPROVED** [review]

**Comment by mischat** [comment]
thanks @hswopeams

**Review by dohaki — APPROVED** [review]
LGTM 👍

**Pull request merged**